### ISSUE TITLE
Object of type <class 'numpy.float64'> cannot be safely interpreted as an integer

### ISSUE THREAD

**Issue by j-rahman** [yt-project/yt#2469]
### Bug report

**Bug summary**

I was trying to learn how to load generic HDF5 data by following the guide at [Loading Generic Array Data](https://yt-project.org/doc/examining/generic_array_data.html) which produces an error when used alongside `Numpy 1.12.0` and newer. It is caused by unsupported float index in these versions. A workaround is to downgrade to `Numpy 1.11.0`.

**How to reproduce**

The last line in the following code using the dataset at http://yt-project.org/data/UnigridData.tar.gz
```python
import yt, h5py, numpy as np
from yt.utilities.physical_ratios import cm_per_kpc
f = h5py.File("test_data_dir/UnigridData/turb_vels.h5", "r") # Read-only access to the file

units = ["gauss","gauss","gauss", "g/cm**3", "erg/cm**3", "K", 
         "cm/s", "cm/s", "cm/s", "cm/s", "cm/s", "cm/s"]

data = {k:(v.value,u) for (k,v), u in zip(f.items(),units)}
bbox = np.array([[-0.5, 0.5], [-0.5, 0.5], [-0.5, 0.5]])
ds   = yt.load_uniform_grid(data, data["Density"][0].shape, length_unit=250.*cm_per_kpc,
       bbox=bbox, nprocs=8, periodicity=(False,False,False))

prj = yt.ProjectionPlot(ds, 'z', ['z-velocity','Temperature','Bx'], weight_field='Density')
```
produces the following error
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/yt-conda/lib/python3.7/site-packages/numpy/core/function_base.py in linspace(start, stop, num, endpoint, retstep, dtype, axis)
    116     try:
--> 117         num = operator.index(num)
    118     except TypeError:

TypeError: 'numpy.float64' object cannot be interpreted as an integer

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-5-2b3dfebd5dea> in <module>
----> 1 prj = yt.ProjectionPlot(ds, 'z', ['z-velocity','Temperature','Bx'], weight_field='Density')

~/yt-conda/lib/python3.7/site-packages/yt/visualization/plot_window.py in __init__(self, ds, axis, fields, center, width, axes_unit, weight_field, max_level, origin, right_handed, fontsize, field_parameters, data_source, method, proj_style, window_size, aspect)
   1481         PWViewerMPL.__init__(self, proj, bounds, fields=fields, origin=origin,
   1482                              right_handed=right_handed, fontsize=fontsize, window_size=window_size,
-> 1483                              aspect=aspect)
   1484         if axes_unit is None:
   1485             axes_unit = get_axes_unit(width, ds)

~/yt-conda/lib/python3.7/site-packages/yt/visualization/plot_window.py in __init__(self, *args, **kwargs)
    661             self._plot_type = kwargs.pop("plot_type")
    662         self._splat_color = kwargs.pop("splat_color", None)
--> 663         PlotWindow.__init__(self, *args, **kwargs)
    664 
    665     def _setup_origin(self):

~/yt-conda/lib/python3.7/site-packages/yt/visualization/plot_window.py in __init__(self, data_source, bounds, buff_size, antialias, periodic, origin, oblique, right_handed, window_size, fields, fontsize, aspect, setup)
    214                 self._field_transform[field] = linear_transform
    215         self.setup_callbacks()
--> 216         self._setup_plots()
    217 
    218     def __iter__(self):

~/yt-conda/lib/python3.7/site-packages/yt/visualization/plot_window.py in _setup_plots(self)
    946                     else: # symlog_transform
    947                         flinthresh = 10**np.floor( np.log10( self.plots[f].cb.norm.linthresh ) )
--> 948                         mticks = self.plots[f].image.norm( get_symlog_minorticks(flinthresh, vmin, vmax) )
    949                     self.plots[f].cax.yaxis.set_ticks(mticks, minor=True)
    950             else:

~/yt-conda/lib/python3.7/site-packages/yt/visualization/plot_container.py in get_symlog_minorticks(linthresh, vmin, vmax)
    148         return np.hstack((-get_log_minorticks(linthresh,-vmin)[::-1], 0) )
    149     else:
--> 150         return np.hstack((-get_log_minorticks(linthresh,-vmin)[::-1], 0,
    151                           get_log_minorticks(linthresh, vmax)))
    152 

~/yt-conda/lib/python3.7/site-packages/yt/visualization/plot_container.py in get_log_minorticks(vmin, vmax)
    120     while cofA*10**expA <= cofB*10**expB:
    121         if expA < expB:
--> 122             lmticks = np.hstack( (lmticks, np.linspace(cofA, 9, 10-cofA)*10**expA) )
    123             cofA = 1
    124             expA += 1

<__array_function__ internals> in linspace(*args, **kwargs)

~/yt-conda/lib/python3.7/site-packages/numpy/core/function_base.py in linspace(start, stop, num, endpoint, retstep, dtype, axis)
    119         raise TypeError(
    120             "object of type {} cannot be safely interpreted as an integer."
--> 121                 .format(type(num)))
    122 
    123     if num < 0:

TypeError: object of type <class 'numpy.float64'> cannot be safely interpreted as an integer.
```

**Version Information**
If this matters at all (which I don't think it does), I am using
  * Operating System: Fedora 31 (Linux 5.5.6-201)
  * Python Version: 3.7.6
  * yt version: 3.5.1
  * Numpy version: 1.18.1 

I have installed `yt` using three independent sources: (1) `pkgs/main::yt-3.4.1-py37h14c3975_1`, (2) `conda-forge::yt-3.5.1-py37hc1659b7_1` and (3) from [All-in-One Installation Script](https://yt-project.org/doc/installing.html#all-in-one-installation-script), and the bug exists regardless.

**Comment by ngoldbaum** [comment]
It looks like this is fixed already in `master` and will be in yt-3.6, see https://github.com/yt-project/yt/pull/2419.

Downgrading to numpy 1.17 should fix this error. If you can't downgrade to numpy 1.17 then building yt from source should also get you the fix.

**Comment by munkm** [comment]
I just verified that this issue does exist with numpy 1.18 but downgrading to 1.17.5 with yt 3.5.1 should resolve the issue.
